yew-ui/src/model/decode/peer_decode_manager.rs

```diff
@@ -0,0 +1,120 @@
+use crate::model::MediaPacketWrapper;
+use std::collections::HashMap;
+use std::sync::Arc;
+use types::protos::media_packet::media_packet::MediaType;
+use types::protos::media_packet::MediaPacket;
+use yew::prelude::Callback;
+
+use super::{AudioPeerDecoder, VideoPeerDecoder};
+
+pub struct MultiDecoder {
+    pub audio: AudioPeerDecoder,
+    pub video: VideoPeerDecoder,
+    pub screen: VideoPeerDecoder,
+}
+
+impl MultiDecoder {
+    fn new(video_canvas_id: String, screen_canvas_id: String) -> Self {
+        Self {
+            audio: AudioPeerDecoder::new(),
+            video: VideoPeerDecoder::new(&video_canvas_id),
+            screen: VideoPeerDecoder::new(&screen_canvas_id),
+        }
+    }
+
+    /// Result is:
+    ///     Ok(media_type, Some(true)) on first frame,
+    ///     Ok(media_type, Some(false)) on other frames
+    ///     Ok(media_type, None) on failure to render
+    ///     Err(e) problem with the packet
+    fn decode(&mut self, packet: &Arc<MediaPacket>) -> Result<(MediaType, Option<bool>), i32> {
```

Comment:
@ronen I think this is awesome! would you please document what the `Option<bool>` is for please?

I got lost a bit myself but is it a flag to indicate if the decode should trigger a re-render?

Reply:
Follow up:  Added a DecodeStatus (though the media_type didn't fit in it)